pkg/notes/notes_map.go

```diff
@@ -0,0 +1,174 @@
+package notes
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+// MapProvider interface that obtains release notes maps from a source
+type MapProvider interface {
+	GetMapsForPR(int) (map[int]*ReleaseNotesMap, error)
+}
+
+// NewProviderFromInitString creates a new map provider from an initialization string
+func NewProviderFromInitString(initString string) (MapProvider, error) {
+	// If init string starts with gs:// return a CloudStorageProvider
+	if initString[0:5] == "gs://" {
+		// Currently for illustration purposes
+		return nil, errors.New("CloudStorageProvider is not yet implemented")
+	}
+
+	// Otherwise, build a DirectoryMapProvider using the
+	// whole init string as the path
+	fileStat, err := os.Stat(initString)
+	if os.IsNotExist(err) {
+		return nil, errors.New("Release notes map path does not exist")
```

Comment:
nit, let's start errors with lower-case chars to ensure proper formatting when wrapped:
```suggestion
		return nil, errors.New("release notes map path does not exist")
```

Reply:
Done, sorry I always forget that one.